utils/evaluate/evaluate.go

```diff
@@ -100,8 +100,8 @@ func EvalCondition(resultValue any, condition string) (bool, error) {
 		"asFloat": asFloat,
 		"isNaN":   math.IsNaN,
 		"isInf":   isInf,
-		"isNil":   isNilFunc(resultValue),
-		"default": defaultFunc(resultValue),
+		"isNil":   isNil,
```

Comment:
Yep, I was baffled by the way it was done because it made no actual sense to me. I had to change it however because this extra level of indirection made it impossible to actually pass a dynamic result vector to the `defaultFunc` or `isNil`.

So with the original function, for a result vector of length 2 (assume non nil for now) and a condition like `default(result[0]) > 10 && default(result[1]) < 4` for example, `defaultFunc` takes the result vector as an argument, and returns the function that just returns this result vector (i.e., it discards the first argument `_ any`). Compiling an expression like the above would result in `default(result[0])` returning the result _vector_, so the comparison ends up being `vector > integer`.

While the new implementation dynamically calls `defaultFunc` during compilation, and it actually considers the first argument, which in our case is the element `result[0]`/`result[1]` passed to it.

Originally, it didn't cause a problem, since it is one value not a vector, and so doing it either way would return the same value.